java-client-model/src/main/java/com/cumulocity/model/authentication/CumulocityOAuthCredentials.java

```diff
@@ -49,16 +70,19 @@ public String getAuthenticationString() {
 
     @Override
     public String getUsername() {
-        String claims = new String(Base64.decodeBase64(oAuthAccessToken.split("\\.")[1]));
-        return Jsons.readField(CUMULOCITY_USER_ID_CLAIM, claims).orNull();
+        return ofNullable(accessTokenClaims.get(CUMULOCITY_USER_ID_CLAIM))
+                .map(String::valueOf)
+                .orElse(null);
     }
 
     @Override
     public String getTenantId() {
-        String[] token = oAuthAccessToken.split("\\.");
-        String header = new String(Base64.decodeBase64(token[0]));
-        String claims = new String(Base64.decodeBase64(token[1]));
-        return Jsons.readField(TENANT_ID_CLAIM, claims).or(Jsons.readField(TENANT_ID_CLAIM, header)).orNull();
+        return Stream.of(accessTokenClaims, accessTokenHeader)
+                .filter(m -> m.containsKey(TENANT_ID_CLAIM))
+                .map(m -> m.get(TENANT_ID_CLAIM))
+                .map(String::valueOf)
+                .findFirst()
+                .orElse(null);
```

Comment:
Optionally you can move this logic to the `parseAccessToken()` and store only `tenantId` and `userId` in object properties instead of 2 maps. Personally I am not sure which one is nicer, and I think AumulocityOAuthCredentials objects are not long living